### DISCUSSION TITLE
characteristic->setValue() if value is the same

### DISCUSSION THREAD

**Discussion by Frtrillo** [h2zero/NimBLE-Arduino#365] [Q&A] [answered]
I wonder if I can do setValue constantly while, lets say with the same value, is wasting CPU time or is attacking the Bluetooth stack? So I can keep doing setValue and it wont do anything until I do a set value with a different value than it had.

Or on the other hand I also wonder if when I do getValue it attacks the bluetooth stack constantly or just read from ram

**Comment by h2zero** [comment, marked as answer]
The call to `setValue()` will just set the value in ram and does not invoke the BLE stack. The same applies to `getValue()`. Only when the peer device issues a read command, or a value notification/indication is sent, will the BLE stack be involved.
